src/persistence/settings.cpp

```diff
@@ -1379,6 +1409,48 @@ void Settings::setCircleExpanded(int id, bool expanded)
     circleLst[id].expanded = expanded;
 }
 
+void Settings::addFriendRequest(const QString &friendAddress, const QString &message)
+{
+    QMutexLocker locker{&bigLock};
+    QPair<QString, QString> request(friendAddress, message);
+
+    if (friendRequests.indexOf(request) != -1)
+        return;
+
+    friendRequests.push_back(request);
+    ++unreadFriendRequests;
+}
+
+unsigned int Settings::getUnreadFriendRequests() const
+{
+    QMutexLocker locker{&bigLock};
```

Comment:
Hm ... not exactly sure, but how about using a `QSharedData` inheritance in combination with `QExplicitlySharedDataPointer`? Isn't that completely thread-safe? If that works, the technique could also be used for the audio data parser.